gto-support-androidx-lifecycle/src/main/java/org/ccci/gto/android/common/androidx/lifecycle/CollectionLiveData.kt

```diff
@@ -16,7 +17,7 @@ sealed class CollectionLiveData<T, C : Collection<T>>(
     fun removeAll(items: Collection<T>) = collection.removeAll(items)
     fun clear() = collection.clear()
 
-    private fun notifyChanged() = postValue(collection as C)
+    private fun notifyChanged() = (collection as C).let { if (synchronous) setValue(it) else postValue(it) }
```

Comment:
What's the difference between ```setValue``` vs ```postValue```?

Reply:
`LiveData.setValue()` immediately updates the LiveData's value and triggers any observers. This can only be called on the main thread.

`LiveData.postValue()` queues up a call to `setValue()` on the main thread that will execute at some point in the future

Reply:
`LiveData.postValue()` can be called from any thread

Reply:
https://developer.android.com/reference/androidx/lifecycle/LiveData#postValue(T)